lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -355,14 +364,56 @@ class HtmlViewEmbedder {
     sceneHost.append(_svgPathDefs!);
   }
 
-  Future<void> submitFrame(CkPicture basePicture) async {
-    final List<CkPicture> pictures = <CkPicture>[basePicture];
-    for (final CkPictureRecorder recorder in _context.pictureRecorders) {
-      pictures.add(recorder.endRecording());
-    }
+  void optimizeRendering() {
+    final Map<CkPicture, PictureLayer> scenePictureToRawPicture =
+        <CkPicture, PictureLayer>{};
+    final Map<CkPictureRecorder, PictureLayer> reversePictureRecorderMap =
+        <CkPictureRecorder, PictureLayer>{};
+    for (final MapEntry<PictureLayer, CkPictureRecorder> entry
+        in _context.pictureRecordersCreatedDuringPreroll.entries) {
+      reversePictureRecorderMap[entry.value] = entry.key;
+    }
+    final List<Object> unoptimizedRendering =
+        _context.sceneElements.map((Object element) {
+      if (element is CkPictureRecorder) {
+        final CkPicture scenePicture = element.endRecording();
+        final PictureLayer rawPicture = reversePictureRecorderMap[element]!;
+        scenePictureToRawPicture[scenePicture] = rawPicture;
+        return scenePicture;
+      } else {
+        return element;
+      }
+    }).toList();
```

Comment:
Can `createOptimizedRendering` simply take an `Iterable` so that you don't have to explicitly turn it into a list?

Reply:
done